src/frontend/src/pages/MyGroup/PendingApplications.jsx

```diff
@@ -1,33 +1,95 @@
-import { Table, TableBody, TableContainer } from '@carbon/react';
-
-import { SmsDataTable, SimpleHeader, SimpleRow } from 'components/SmsDataTable';
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import AppRoutes from 'navigation/AppRoutes';
+import {
+    Table,
+    TableBody,
+    TableContainer,
+    TableCell,
+    TableRow,
+    TableBatchActions, 
+    TableBatchAction, 
+    TableToolbar,
+    TableSelectRow,
+    TableHeader,
+    TableHead,
+    TableSelectAll
+} from '@carbon/react';
+import { Close, Checkmark } from '@carbon/react/icons';
+import { SmsDataTable, SimpleHeader } from 'components/SmsDataTable';
+import { useManageGroupApplicaton } from 'hooks';
 
 export function PendingApplications(groupApplications) {
+    const { manageGroupApplicaton } = useManageGroupApplicaton();
+    const navigate = useNavigate();
     const headers = [
         { key: 'name', header: 'Name', isSortable: true },
         { key: 'message', header: 'Message' }
-        // TODO: add management logic
-        // { key: 'options', header: 'Options' }
     ];
     const applications = groupApplications.groupApplications;
 
+    const batchActionClick = (selectedRows, status) => () =>
+        {
+            manageGroupApplicaton({
+                ids: selectedRows.filter(data => data.hasOwnProperty('id')).map(data => data.id),
+                status: status});
+            window.location.reload(false);
+        }
+
     return (
         <SmsDataTable rows={applications} headers={headers} className='groups-page__datatable'>
-            {({ rows, headers, getHeaderProps, getTableProps }) => (
+            {({ rows, headers, getHeaderProps, getTableProps, getSelectionProps, getRowProps, getToolbarProps, selectedRows, getBatchActionProps }) => {
+                const batchActionProps = getBatchActionProps(); 
+                
+                return (
+                
                 <TableContainer
                     title='Pending Group Application'
                     description='This is a list of all the group application for this semester.'
                 >
+                    <TableToolbar {...getToolbarProps()}>
+                        <TableBatchActions {...batchActionProps}>
+                            <TableBatchAction
```

Comment:
Keeping in mind here, it doesn't make sense that the group accepts invitations. Rather a student would accept an invitation from a group. We should remove this logic from here.

Rejecting makes perfect since here.